rust/src/dns/parser.rs

```diff
@@ -1120,21 +1119,7 @@ mod tests {
 
     #[test]
     fn test_dns_parse_name_truncated_max_segments_with_pointer() {
-        let mut buf: Vec<u8> = vec![];
-
-        // "a" at the beginning of the buffer.
-        buf.push(0b0000_0001);
-        buf.push(b'a');
-
-        // Followed by a pointer back to the beginning.
-        buf.push(0b1100_0000);
-        buf.push(0b0000_0000);
-
-        // The start of the name, which is pointer to the beginning of
-        // the buffer.
-        buf.push(0b1100_0000);
-        buf.push(0b000_0000);
-
+        let buf: Vec<u8> = vec![0b0000_0001, b'a', 0b1100_0000, 0b0000_0000, 0b1100_0000, 0b000_0000];
```

Comment:
This loses the documentation of the test. It would be better to suppress the warning here I think.